mongomock/aggregate.py

```diff
@@ -634,6 +635,62 @@ def _handle_date_operator(self, operator, values):
                     ' in Mongomock.'
                 )
             return out_value['date'].strftime(out_value['format'])
+        if operator == '$dateFromParts':
+            if not isinstance(values, dict):
+                raise OperationFailure(
+                    '$dateFromParts operator must correspond a dict '
+                    'that has "year" or "isoWeekYear" field.'
+                )
+            if (
+                not isinstance(values, dict)
```

Comment:
You've already checked for that above and bailed out, so no need to check again.